### PR TITLE
feat(evm): support multiple languages for deriveKey cheatcode

### LINKED ISSUE #4560: unable to parse mnemonic in French
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (3a15fe9 2023-01-29T00:11:51.291099Z)

### What command(s) is the bug in?

forge test

### Operating System

macOS (Intel)

### Describe the bug

Hello,

using a mnemonic like:
`rugueux véloce régulier prairie ....`

running following solidity code:
```solidity
string memory mnemonic = vm.envString("MNEMONIC");
address deployer = vm.addr(vm.deriveKey(mnemonic, 0));
        
```

I get the following : 

```solidity 
[FAIL. Reason: the word `rugueux` is invalid] testAccount() (gas: 7518)
Traces:
  [7518] Mangrove::testAccount() 
    ├─ [0] VM::selectFork(0) 
    │   └─ ← ()
    ├─ [0] VM::envString(MNEMONIC) [staticcall]
    │   └─ ← <env var value>
    ├─ [0] VM::deriveKey(<pk>) [staticcall]
    │   └─ ← "the word `rugueux` is invalid"
    └─ ← "the word `rugueux` is invalid"

````

Is there a way to specify language pack to use ?

FYI tested with the same mnemonic and ethersjs/hardhat, is working fine (despite ethersjs may require to specify the language pack)

### PR DESCRIPTION
**Pull request by 0xYYY**
## Motivation

Resolves #4560.

## Solution

Add support for other languages' wordlist in https://github.com/summa-tx/bitcoins-rs/pull/119/.

### CODE REVIEW

File: evm/src/executor/inspector/cheatcodes/util.rs [hunk 1]
```diff
@@ -146,9 +152,41 @@ pub fn apply<DB: Database>(
         HEVMCalls::Addr(inner) => addr(inner.0),
         HEVMCalls::Sign(inner) => sign(inner.0, inner.1.into(), data.env.cfg.chain_id),
         HEVMCalls::DeriveKey0(inner) => {
-            derive_key(&inner.0, DEFAULT_DERIVATION_PATH_PREFIX, inner.1)
+            derive_key::<English>(&inner.0, DEFAULT_DERIVATION_PATH_PREFIX, inner.1)
         }
-        HEVMCalls::DeriveKey1(inner) => derive_key(&inner.0, &inner.1, inner.2),
+        HEVMCalls::DeriveKey1(inner) => derive_key::<English>(&inner.0, &inner.1, inner.2),
+        HEVMCalls::DeriveKey2(inner) => match inner.2.as_str() {
```
Comment (mattsse) [inline evm/src/executor/inspector/cheatcodes/util.rs:158]: I'd like to have this as standalone function `derive_key_with_wordlist` that accepts the wordlist as str. perhaps even worth having an enum for this

**Review by mattsse — CHANGES_REQUESTED** [review]
nice!

nit re str matching

**Comment by prestwich** [comment]
`coins-bip39@0.8.4` contains non-english wordlists. Thanks @0xYYY!

**Comment by 0xYYY** [comment]
pending another fix https://github.com/summa-tx/coins/pull/120

File: Cargo.toml [hunk 2]
```diff
@@ -60,15 +60,15 @@ codegen-units = 1
...
-ethers-solc = { git = "https://github.com/gakonst/ethers-rs", rev = "80ac394" }
+ethers = { git = "https://github.com/0xYYY/ethers-rs", branch = "chore/bump-bitcoin-rs" }
+ethers-addressbook = { git = "https://github.com/0xYYY/ethers-rs", branch = "chore/bump-bitcoin-rs" }
+ethers-core = { git = "https://github.com/0xYYY/ethers-rs", branch = "chore/bump-bitcoin-rs" }
+ethers-contract = { git = "https://github.com/0xYYY/ethers-rs", branch = "chore/bump-bitcoin-rs" }
+ethers-providers = { git = "https://github.com/0xYYY/ethers-rs", branch = "chore/bump-bitcoin-rs" }
+ethers-signers = { git = "https://github.com/0xYYY/ethers-rs", branch = "chore/bump-bitcoin-rs" }
+ethers-middleware = { git = "https://github.com/0xYYY/ethers-rs", branch = "chore/bump-bitcoin-rs" }
+ethers-etherscan = { git = "https://github.com/0xYYY/ethers-rs", branch = "chore/bump-bitcoin-rs" }
+ethers-solc = { git = "https://github.com/0xYYY/ethers-rs", branch = "chore/bump-bitcoin-rs" }
```
Comment (DaniPopes) [inline Cargo.toml:71]: We don't depend on ethers git anymore, can this be in a new release?

**Comment by Evalir** [comment]
hey @0xYYY — do you plan on getting back to this soon?

**Comment by 0xYYY** [comment]
> hey @0xYYY — do you plan on getting back to this soon?

yeah, plan to finish this around next week

**Comment by 0xYYY** [comment]
added a standalone `derive_key_with_wordlist ` function, bumped ethers version, rebased on master, all tests have passed, ready for review

cc @Evalir @mattsse

**Review by Evalir — APPROVED** [review]
this lgtm—deferring to matt for final approval. we could maybe add a simple failing test to ensure it fails as expected?

**Comment by 0xYYY** [comment]
> hmm seems we have some tests failing?

yeah, but weirdly can't see any details about why the tests failed

Screenshot 2023-05-10 at 23 01 38

**Comment by 0xYYY** [comment]
> hmm seems we have some tests failing?

seems like the failing tests have ben re-ran and all tests have passed now

**Comment by Evalir** [comment]
yup guessed they were just outage/unrelated failures, nice! will wait for matt to merge—nice work :)

**Comment by Evalir** [comment]
hey @0xYYY — do you think we could rebase this? sorry for the extra work, but we recently refactored the abigen part of cheatcodes so will need to redo that bit

**Comment by 0xYYY** [comment]
> hey @0xYYY — do you think we could rebase this? sorry for the extra work, but we recently refactored the abigen part of cheatcodes so will need to redo that bit

no prob, on it

**Review by Evalir — APPROVED** [review]
aight this seems ok now—just seems we have an unrelated ci failure due to a flaky test

**Comment by 0xYYY** [comment]
>  will wait for matt to review/merge—nice work :)

hi @mattsse, pls let me know if there are any changes that i can make to push forward this pr, thx

**Review by Evalir — APPROVED** [review]
This is good, but will need a rebase & a redo on the abi crate prob. will do so tomorrow and get this over the line cc @mattsse

**Comment by 0xYYY** [comment]
thanks @Evalir for pushing this over the line

**Pull request merged**